### PR TITLE
Always output all packages in build info

### LINKED ISSUE #1124: Selecting a package causes code to fail to compile due to invalid `BuildInfo.purs` file
I ran `npx spago test -p spago` so as to skip the `docs-search` tests, and got the following error:
```
npx spago test -p spago
Refreshing the Registry Index...
Reading Spago workspace configuration...
Read the package set from the registry

✅ Selecting package to build: spago

Downloading dependencies...
Building...
[1 of 9] Compiling Spago.Generated.BuildInfo
[2 of 9] Compiling Spago.BuildInfo
[1/1 PropertyIsMissing] src/Spago/BuildInfo.purs:78:23

  78  currentSpagoVersion = BuildInfo.buildInfo.packages."spago-bin"
                            ^^^^^^^^^^^^^^^^^^^

  Type of expression lacks required label "spago-bin".
  while matching label packages
  while checking that type { packages :: { spago :: String
                                         }
                           , pursVersion :: String
                           , spagoVersion :: String
                           }
    is at least as general as type { packages :: { "spago-bin" :: String
                                                 | t0
                                                 }
                                   | t1
                                   }
  while checking that expression buildInfo
    has type { packages :: { "spago-bin" :: String
                           | t0
                           }
             | t1
             }
  while checking type of property accessor buildInfo.packages
  while checking type of property accessor (buildInfo.packages)."spago-bin"
  in value declaration currentSpagoVersion
  where t0 is an unknown type
        t1 is an unknown type

           Src   Lib   All
Warnings     0     0     0
Errors       1     0     1

❌ Failed to build.
```

### PR DESCRIPTION
**Pull request by JordanMartinez**
### Description of the change

Fixes #1124. I don't think there's any reason not to write out all packages... Does this need a test?

### Checklist:

- [ ] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [ ] Added a test for the contribution (if applicable)

**P.S.**: the above checks are not compulsory to get a change merged, so you may skip them. However, taking care of them will result in less work for the maintainers and will be much appreciated 😊

### CODE REVIEW

**Comment by f-f** [comment]
Ah brilliant, this is a good fix

I would say this doesn't need a test, but as you noted in the issue thread this thing has already regressed twice, which is strongly hinting at needing a test 😄

**Comment by JordanMartinez** [comment]
Yeah.... I guess it does need a test.

Besides adding such tests, I also added an inline directive for `buildInfo`. On a separate note, is there a reason why `buildInfo` is a record containing each value rather than having each field within that record be a top-level value? (e.g. `BuildInfo.packages.foo` rather than `BuildInfo.buildInfo.packages.foo`)?

**Comment by JordanMartinez** [comment]
> > On a separate note, is there a reason why buildInfo is a record containing each value rather than having each field within that record be a top-level value? (e.g. BuildInfo.packages.foo rather than BuildInfo.buildInfo.packages.foo)?
> 
> Yes - I don't recall if we have TODOs about this anywhere, but it has that shape so that we can include things like build time, git commit and other non-package-specific info

Just to clarify, you are saying there is a reason for the current code below:
```purs
module Spago.Generated.BuildInfo where

buildInfo :: { packages :: <recordType>, pursVersion :: String, spagoVersion :: String }
buildInfo =
  { packages: { "foo": "1.2.3", "bar": "1.2.3" }
  , pursVersion: "1.2.3"
  , spagoVersion: "1.2.3"
  }
```

rather than this alternative implementation:
```purs
module Spago.Generated.BuildInfo where

packages :: <recordType>
packages =
  { "foo": "1.2.3"
  , "bar": "1.2.3"
  }

pursVersion :: String
pursVersion = "1.2.3"

spagoVersion :: String
spagoVersion = "1.2.3"
```

Could you clarify what that reason is?

**Comment by f-f** [comment]
Oh, I didn't get what you meant then - no specific reason for that, they are pretty much equivalent. If we want to change this we should do it asap, because I won't want to change it liberally once we are out of alpha.

File: test/Prelude.purs [hunk 1]
```diff
@@ -60,9 +63,15 @@ withTempDir = Aff.bracket createTempDir cleanupTempDir
 
       spago = spago' StdinNewPipe
 
+      pursVersion = do
+        let pursBinary = if Process.platform == Just Win32 then "purs.cmd" else "purs"
+        Cmd.exec pursBinary [ "--version" ]
+          $ Cmd.defaultExecOptions { pipeStdout = false, pipeStderr = false, pipeStdin = StdinNewPipe }
+
```
[the remark this answers is not in the document]
Comment (JordanMartinez) [reply test/Prelude.purs]: Done.

File: test/Spago/Build.purs [hunk 2]
```diff
@@ -10,6 +10,7 @@ import Spago.Core.Config as Config
 import Spago.FS as FS
 import Test.Spago.Build.Pedantic as Pedantic
 import Test.Spago.Build.Polyrepo as BuildPolyrepo
+import Test.Spago.Build.BuildInfo as BuildBuildInfo
```
[the remark this answers is not in the document]
Comment (JordanMartinez) [reply test/Spago/Build.purs]: I was following the convention used in `BuildPolyRepo` :smile:

**Review by f-f — APPROVED** [review]

**Pull request merged**